File: 03-frontend/angular-capstone/src/app/common/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { OrderItem } from "./order-item";

export class Purchase {

    customer: Customer;
    shippingAddress: Address;
    billingAddress: Address;
    order: Order;
    orderItems: OrderItem[];

        constructor() {
            this.customer = new Customer();
            this.shippingAddress = new Address();
            this.billingAddress = new Address();
            this.order = new Order();
            this.orderItems = [];
        }
}

File: 02-backend/spring-boot-capstone/src/main/java/com/capstone/ecommerce/dto/Purchase.java
package com.capstone.ecommerce.dto;

import com.capstone.ecommerce.entity.Address;
import com.capstone.ecommerce.entity.Customer;
import com.capstone.ecommerce.entity.Order;
import com.capstone.ecommerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}
